templates/Audio/WAV.tcl

```diff
@@ -1,38 +1,224 @@
+        if {$coding_history_length > 0} {
+            ascii $coding_history_length "Coding History"
+        } else {
+            entry "Coding History" "(Not Present)"
+        }
+    }
 }
+
+proc parse_fact {} {
+    uint32 "Sample Count"
+}
+
+proc parse_chunk {signature length} {
+    entry "Chunk Contents" "" $length
+    set content_start [pos]
+    switch $signature {
+        "fmt " { parse_fmt }
+        "PEAK" { parse_peak }
+        "bext" { parse_bext $length}
+        "fact" { parse_fact }
+        "cue " { parse_cue }
+        "labl" { parse_labl $length}
+        "ltxt" { parse_ltxt $length}
+    }
+    goto [expr $content_start + $length + ($length % 2)]
+}
+
+proc parse_list {length ds64} {
+    section "Chunk List" {
+        set remain $length
+        set index 0
+        while {$remain > 0} {
+            section $index {
+                set chunk_signature [ascii 4 "Chunk Signature"]
+                
+                if {[dict exists $ds64 $chunk_signature]} {
+                    uint32 ;# Will be 0xFFFFFFFF
+                    set chunk_size [dict get $ds64 $chunk_signature]
+                    entry "Chunk Size DS64" $chunk_size
+                } else {
+                    set chunk_size [uint32 "Chunk Size"]
+                }
+
+                set chunk_displancement [expr $chunk_size + $chunk_size % 2 ]
+
+                if {$chunk_signature == "LIST"} {
+                    ascii 4 "LIST Form"
+                    parse_list [expr $chunk_size - 4] $ds64
+                } else {
+                    parse_chunk $chunk_signature $chunk_size
+                }
+            }
+            incr index
+            set remain [expr $remain - ($chunk_displancement + 8) ]
+        }
+    }
+}
+
+proc parse_rf64 {} {
+    section "RF64 Extended Header" {
+        ascii 4 "DS64 Header Signature"
+        set ds64_length [uint32 "DS64 Header Size"]
+        set ds64_start [pos]
+
+        set rf64_size [uint64 "Form Size"]
+        set data_size [uint64 "Data Chunk Size"]
+        uint64; # dead value, historically was frame count
+        
+        set ds64_dict [dict create "data" $data_size]
+        set count [uint32 "Long Chunk Table Length"]
+
+        for {set i 0} {$i < $count} {incr i}  {
+            dict set $ds64_dict [ascii 4 "Long Chunk ID"] [uint64 "Long Chunk Size"]
+        }
+    }
+    goto [expr $ds64_start + $ds64_length]
+    parse_list [expr $rf64_size - ( 12 + $ds64_length )] $ds64_dict
+}
+
+proc parse_wave length {
+    set remain [expr $length - 4]
+    set empty_ds64 [dict create]
+    parse_list $remain $empty_ds64
+}
+
+little_endian
+requires 8 "57 41 56 45" ;# WAVE
+set header_signature [ascii 4 "Header Signature"]
+set riff_size [uint32 "RIFF Size"]
+set riff_form [ascii 4 "RIFF Form"]
+if {$header_signature == "RF64" || $header_signature == "BW64"} {
+    parse_rf64 
+} elseif {$header_signature == "RIFF" } {
+    parse_wave $riff_size
+}
```

Comment:
nit: add ending new line.

I also wonder if it should error out here in an ending `else` condition if the signature is unsupported? Can we use the [error](https://www.tcl.tk/man/tcl8.6/TclCmd/error.htm) command and have that message show in HF (which should handle generic script errors)?

Reply:
The requested newline has been added.

An `error` command would be useful in many places, particularly if it captured the current file pos and reported it.

Reply:
Are you able to try out the existing TCL error command I linked above?